tests/serve/mock/faker/test_stateful_faker.py

```diff
@@ -0,0 +1,414 @@
+    assert (
+        "val1"
+        == mock_data_store["default"].item[res.bodyAsJson["item"]["itemId"]]["bar"]
+    )
+    assert (
+        30 == mock_data_store["default"].item[res.bodyAsJson["item"]["itemId"]]["foo"]
+    )
+
+
+def test_sateless_faker_1(mock_data_store):
+    faker = StatefulFaker(mock_data_store)
+
+    request = RequestBuilder.from_dict(
+        dict(method="get", protocol="http", path="/", host="api.com")
+    )
+
+    schema = {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "required": ["foo", "baz"],
+            "properties": {
+                "foo": {"type": "number"},
+                "bar": {"type": "string"},
+                "baz": {"type": "string"},
+            },
+        },
+    }
+    res = faker.process(
+        OpenAPISpecification(source="default", api=spec(response_schema=schema)),
+        request,
+    )
+
+    assert valid_schema(res.bodyAsJson, schema)
+
+
+def test_sateless_faker_2(mock_data_store):
+    faker = StatefulFaker(mock_data_store)
+
+    request = RequestBuilder.from_dict(
+        dict(method="get", protocol="http", path="/", host="api.com")
+    )
+
+    schema = {
+        "$id": "https://example.com/person.schema.json",
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "title": "Person",
+        "type": "object",
+        "properties": {
+            "firstName": {"type": "string", "description": "The person's first name."},
+            "lastName": {"type": "string", "description": "The person's last name."},
+            "age": {
+                "description": "Age in years which must be equal to or greater than zero.",
+                "type": "integer",
+                "minimum": 0.0,
+            },
+        },
+    }
+    res = faker.process(
+        OpenAPISpecification(source="default", api=spec(response_schema=schema)),
+        request,
+    )
+
+    assert valid_schema(res.bodyAsJson, schema)
+
+
+def test_sateless_faker_3(mock_data_store):
+    faker = StatefulFaker(mock_data_store)
+
+    request = RequestBuilder.from_dict(
+        dict(method="get", protocol="http", path="/", host="api.com")
+    )
+
+    schema = {
+        "$id": "https://example.com/geographical-location.schema.json",
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "title": "Longitude and Latitude Values",
+        "description": "A geographical coordinate.",
+        "required": ["latitude", "longitude"],
+        "type": "object",
+        "properties": {
+            "latitude": {"type": "number", "minimum": -90.0, "maximum": 90.0},
+            "longitude": {"type": "number", "minimum": -180.0, "maximum": 180.0},
+        },
+    }
+    res = faker.process(
+        OpenAPISpecification(source="default", api=spec(response_schema=schema)),
+        request,
+    )
+
+    assert valid_schema(res.bodyAsJson, schema)
+
+
+def test_sateless_faker_4(mock_data_store):
```

Comment:
I know it's hard to name tests but something saying what is actually being tested in each test would be really helpful! Also some of the tests are really big, I think smaller ones might be easier to debug.

Reply:
I don't know what is tested inside. I didn't write these tests, just copied from an old faker.